ts/types/Attachment.ts

```diff
@@ -34,3 +43,59 @@ export const isVisualMedia = (attachment: Attachment): boolean => {
   const isSupportedVideoType = GoogleChrome.isVideoTypeSupported(contentType);
   return isSupportedImageType || isSupportedVideoType;
 };
+
+export const save = ({
+  attachment,
+  document,
+  getAbsolutePath,
+  timestamp,
+}: {
+  attachment: Attachment;
+  document: Document;
+  getAbsolutePath: (relativePath: string) => string;
+  timestamp?: number;
+}): void => {
+  const url = !is.undefined(attachment.path)
+    ? getAbsolutePath(attachment.path)
+    : arrayBufferToObjectURL({
+        data: attachment.data,
+        type: SAVE_CONTENT_TYPE,
+      });
+  const filename = getSuggestedFilename({ attachment, timestamp });
+  saveURLAsFile({ url, filename, document });
+  URL.revokeObjectURL(url);
```

Comment:
What happens if you call `URL.revokeObjectUrl()` on a path to a file attachment on disk?

Reply:
Good catch 🎣 I’ll make it guarded to be safe.